### DISCUSSION TITLE
Foreign constant violation error (usergroups_groupid_foreign) logging in with GitHub

### DISCUSSION THREAD

**Discussion by pferreir** [requarks/wiki#3667] [Error / Bug Report]
**Describe the bug**
When some of my users log in through GitHub auth, they get a foreign constant violation error (`usergroups_groupid_foreign`).

**To Reproduce**
1. Add a group under "Assign to group" in your GitHub authentication settings
![image](https://user-images.githubusercontent.com/2699/80082588-dd041200-8554-11ea-9740-ed212e6cafb2.png)
2. Ask user to log in for the first time, using GitHub
3. An error pops up (see screenshot below)
4. The user is apparently still added to the DB, but not to the group

From what I saw, erasing the group from the settings and adding it back seems to fix it at least temporarily. But the problem keeps coming back, for some reason.

**Expected behavior**
There shouldn't be an error, the user should be added to the group.

**Screenshots**
![image](https://user-images.githubusercontent.com/2699/80082478-bba32600-8554-11ea-9e00-6bada46fd0b8.png)

**Host Info (please complete the following information):**
 - OS: Official (OpenShift) Docker image
 - Wiki.js version: 2.2.5.1
 - Database engine: postgres 11.4

**Comment by Saicheg** [comment]
👍 for this issue, exact same error for Google and wiki.js 2.4 version

**Comment by Saicheg** [comment]
@pferreir i would love to help this problem to be solved, what is the the best way to help you with that?

**Comment by pferreir** [comment]
Hi @Saicheg, I'm not a project maintainer or anything. Just a regular user. Like you, I'll be glad to provide any information.

**Comment by Saicheg** [comment]
Ha @pferreir, sorry about that, tagged a wrong one. It was a question to @NGPixel i guess

**Comment by bobs-bg** [comment]
hey guys is there any solution on this one?

**Reply by Saicheg** [reply]
@bobs-bg don't think so, we are manually assigning roles to every new user while onboarding :)

**Comment by timothystewart6** [comment]
This happened to me too on a clean installation right after changing the default group that new users are assigned to. 

I changed from  `Guest` to another group. Now, no matter which group I choose all new users see this error after creating an account until I manually assign them. I feel like this is a bug.
